File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def check(self, ch: str) -> bool :
        if ch.isupper() : ch = ch.lower()
        return ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u'
    def halvesAreAlike(self, s: str) -> bool:
        ans = 0
        for i, ch in enumerate(s) :
            if i < len(s)//2 and self.check(ch): ans += 1
            if i >= len(s)//2 and self.check(ch): ans -= 1
        return ans == 0
        

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool check(char ch) {
        if(ch <= 'Z') ch += 32;
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }
    bool halvesAreAlike(string s) {
        int ans = 0;
        for(int i=0; i<s.size(); i++) {
            if(i < s.size()/2 && check(s[i])) ans++;
            else if(i >= s.size()/2 && check(s[i])) ans--;
        }
        return ans == 0;
    }
};